### PR TITLE
[GUI] tab on TextInputs

### PR DESCRIPTION
**Pull request by pythcoiner**
this PR  introduce handling of new events in all views:
- `Tab` will move cursor to the next TextInput
- `Tab` + `Shift` will move the cursor to previous TextInput

### CODE REVIEW

File: gui/src/main.rs [hunk 1]
```diff
@@ -279,11 +317,13 @@ impl Application for GUI {
                 State::Launcher(v) => v.subscription().map(|msg| Message::Launch(Box::new(msg))),
             },
             iced_native::subscription::events().map(Self::Message::Event),
+            self.key_subscription(),
```
Comment (edouardparis) [inline gui/src/main.rs:320]: self.key_subscription is using iced_native::subscriptions::events_with, which is the same subscription that the one line 319. Please merge them together like this:

```rust
          iced_native::subscription::events_with(|event, status| match (&event, status) {
                 (
                    Event::Keyboard(keyboard::Event::KeyPressed {
                        key_code: KeyCode::Tab,
                        modifiers,
                        ..
                    }),
                    event::Status::Ignored,
                ) => Some(Message::KeyPressed(Key::Tab(modifiers.shift()))),
                (
                    iced::Event::Window(iced_native::window::Event::CloseRequested),
                    event::Status::Ignored,
                ) => Some(Message::Event(event)),
                _ => None,
            }),
```

**Review by edouardparis — COMMENTED** [review]
Very interesting :eyes:  Thanks for looking for cool feature like this in the iced examples

File: gui/src/main.rs (hunk 1 above)
Comment (pythcoiner) [reply gui/src/main.rs]: done

**Review by edouardparis — APPROVED** [review]
ACK 158bdc8a70bcd1a5dae34a842e7ac0bb39c20127

**Pull request merged**